libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -98,7 +100,15 @@ internal class MapboxTripSession(
             is SetRefreshedRoutesInfo -> {
                 if (routes.isNotEmpty()) {
                     val primaryRoute = routes.first()
-                    navigator.refreshRoute(primaryRoute).fold(
+                    var refreshRouteResult: Expected<String, List<RouteAlternative>>? = null
+                    for (route in routes) {
+                        refreshRouteResult = navigator.refreshRoute(route)
+                    }
+                    if (refreshRouteResult!!.isValue) {
+                        this@MapboxTripSession.primaryRoute = routes.first()
+                        roadObjects = primaryRoute.upcomingRoadObjects
+                    }
+                    refreshRouteResult.fold(
```

Comment:
it looks confuse. Why is stored only the latest result from, where every `Navigator#rerfresRoute` return its own result 🤔

Reply:
- what happens if the primary route refresh fails, but alternatives succeed?
- what if one of the alternatives failed and the other passed success?
probably both cases are fine, but just to clarify

could you leave a `todo` why it's happening and cut a ticket for NN to change API  

```txt
// the very last result contains the whole required data
// TODO API change request #NN{TICKET_NUMBER}
```

Reply:
Ticket: NN-110

> what happens if the primary route refresh fails, but alternatives succeed?
what if one of the alternatives failed and the other passed success?
probably both cases are fine, but just to clarify

That's what I'm trying to figure out [here](https://github.com/mapbox/mapbox-navigation-android/pull/6389#discussion_r980284727). It seems like everyone agrees we should remember the last successful result. Will update the PR.

> it looks confusing. Why is stored only the latest result from, where every Navigator#rerfresRoute return its own result 🤔

The last one = the actual one, but that only works in case when everything is successful.

Reply:
could you inject `todo NN-110` into the code to address anybody who would like to know why it was done so

Reply:
Sure.